src/Symfony/Component/Form/ResolvedFormType.php

```diff
@@ -45,11 +50,18 @@ class ResolvedFormType implements ResolvedFormTypeInterface
 
     public function __construct(FormTypeInterface $innerType, array $typeExtensions = array(), ResolvedFormTypeInterface $parent = null)
     {
-        if (!preg_match('/^[a-z0-9_]*$/i', $innerType->getName())) {
+        $name = $innerType->getName();
+
+        // As of Symfony 2.8, getName() returns null by default. In this case
+        // (and as of Symfony 3.0), the FQCN of the type is used as name
+        // The deprecation error is triggered in FormRegistry
+        if (null === $name) {
+            $name = get_class($innerType);
+        } elseif (!preg_match('/^[a-z0-9_]*$/i', $name)) {
```

Comment:
yeah, but the fact that we allow FQCNs means that chars allowed in them are now allowed in type names.

and if someone wants to create a new form type now compatible with Symfony 2.3 to 3.0, the right way would be to use the FQCN already (so that anyone using this type could use the same code regarding their Sf version). This would be done by returning the FQCN from ``getName()``, which is not accepted (actually, I'm thinking that it would still not work as older versions would reject such type name, so there might be no way to write new form types in a way working both on 2.3 and 3.0, while we managed to do it until now)